clients/egui/src/onboard.rs

```diff
@@ -0,0 +1,335 @@
+            create_err: None,
+            acct_str: String::new(),
+            import_err: None,
+            import_status: None,
+        }
+    }
+
+    pub fn update(&mut self, ctx: &egui::Context) -> Option<OnboardHandOff> {
+        let mut resp = None;
+
+        while let Ok(update) = self.update_rx.try_recv() {
+            match update {
+                Update::AccountCreated(result) => match result {
+                    Ok(acct_data) => {
+                        resp = Some(OnboardHandOff {
+                            settings: self.settings.clone(),
+                            core: self.core.clone(),
+                            acct_data,
+                        });
+                    }
+                    Err(msg) => {
+                        self.state = State::Idle(Route::Create);
+                        self.create_err = Some(msg);
+                    }
+                },
+                Update::AccountImported(maybe_err) => {
+                    if let Some(msg) = maybe_err {
+                        self.state = State::Idle(Route::Import);
+                        self.import_err = Some(msg);
+                    }
+                }
+                Update::ImportSyncProgress(sp) => {
+                    self.import_status = Some(match &sp.current_work_unit {
+                        lb::ClientWorkUnit::PullMetadata => "Pulling file tree updates".to_string(),
+                        lb::ClientWorkUnit::PushMetadata => "Pushing file tree updates".to_string(),
+                        lb::ClientWorkUnit::PullDocument(name) => format!("Pulling: {}", name),
+                        lb::ClientWorkUnit::PushDocument(name) => format!("Pushing: {}", name),
+                    });
+                }
+                Update::ImportSyncDone(maybe_err) => {
+                    if let Some(err) = maybe_err {
+                        match err {
+                            SyncError::Major(msg) => self.import_err = Some(msg),
+                            SyncError::Minor(msg) => self.import_err = Some(msg),
+                        }
+                    } else {
+                        self.import_status = Some("Loading account data...".to_string());
+                    }
+                }
+                Update::AccountDataLoaded(result) => match result {
+                    Ok(acct_data) => {
+                        resp = Some(OnboardHandOff {
+                            settings: self.settings.clone(),
+                            core: self.core.clone(),
+                            acct_data,
+                        });
+                    }
+                    Err(err) => self.import_err = Some(err),
+                },
+            }
+        }
+
+        egui::CentralPanel::default().show(ctx, |ui| {
+            ui.vertical_centered(|ui| {
+                ui.set_max_width(400.0);
+
+                ui.add_space(40.0);
+                self.logo.show_scaled(ui, 1.0);
```

Comment:
I feel like the logo shouldn't be translucent and should follow the other lockbook clients by having a solid one.

Reply:
Have we chosen one specific logo? Or just any non-transparent one will do?

Reply:
It looked much worse than this to me, but I still kept it around in the client directory to easily switch back.

Reply:
I think every client uses a different variation of the logo. So, in my opinion, any is fine, but maybe @Parth wants to chime in.

Reply:
yeah we should unify, but not in this pr